### DISCUSSION TITLE
How to attach file to data form

### DISCUSSION THREAD

**Discussion by Liqynn** [serenity-is/Serenity#5889] [Help] [answered]
Hello
  How to attach file to data form ? 
i.e. how to attach invoice scan to invoice data form ?

**Comment by edwardch** [comment, marked as answer]
Add `[FileUploadEditor]` to the field on your XyzForm.cs

**Reply by dfaruque** [reply]
`[FileUploadEditor]` should always be used in the Row.cs file. Otherwise the file will be saved inside temp folder!

**Comment by Liqynn** [comment]
Thanks.
Is this control presents on https://serenity.is/demo   ?

**Reply by edwardch** [reply]
See https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.Northwind/Serenity.Demo.Northwind/Product/ProductRow.cs

Yes it's used similarly to `ImageUploadEditor`

**Reply by edwardch** [reply]
@Liqynn if this is your answer please mark it as answered. It really helps others in future when they face similar issues to yours. 
Thanks!

**Reply by Liqynn** [reply]
Done
